internal/v1alpha2/controllers/bundledeployment/bundledeployment.go

```diff
@@ -0,0 +1,361 @@
+		if err != nil {
+			setDynamicWatchFailed(&bundleDeployment.Status.Conditions, err.Error(), bundleDeployment.Generation)
+			return ctrl.Result{}, err
+		}
+
+		unstructuredObj := &unstructured.Unstructured{Object: uMap}
+		if err := func() error {
+			b.dynamicWatchMutex.Lock()
+			defer b.dynamicWatchMutex.Unlock()
+
+			_, isWatched := b.dynamicWatchGVKs[unstructuredObj.GroupVersionKind()]
+			if !isWatched {
+				if err := b.controller.Watch(
+					&crsource.Kind{Type: unstructuredObj},
+					&handler.EnqueueRequestForOwner{OwnerType: bundleDeployment, IsController: true},
+					helmpredicate.DependentPredicateFuncs()); err != nil {
+					return err
+				}
+				b.dynamicWatchGVKs[unstructuredObj.GroupVersionKind()] = struct{}{}
+			}
+			return nil
+		}(); err != nil {
+			setDynamicWatchFailed(&bundleDeployment.Status.Conditions, err.Error(), bundleDeployment.Generation)
+			return ctrl.Result{}, err
+		}
+	}
+	return ctrl.Result{}, nil
+}
+
+// unpackContents unpacks contents from all the sources, and stores under a directory referenced by the bundle deployment name.
+// It returns the consolidated state on whether contents from all the sources have been unpacked.
+func (b *bundleDeploymentReconciler) unpackContents(ctx context.Context, bundledeployment *v1alpha2.BundleDeployment, store store.Store) (*source.Result, error) {
+	unpackResult := make([]source.Result, 0)
+
+	// unpack each of the sources individually, and consolidate all their results into one.
+	for _, src := range bundledeployment.Spec.Sources {
+		res, err := b.unpacker.Unpack(ctx, src, store, source.UnpackOption{
+			BundleDeploymentUID: bundledeployment.UID,
+		})
+		if err != nil {
+			return nil, err
+		}
+		unpackResult = append(unpackResult, *res)
+	}
+	// Even if one source has not unpacked, update Bundle Deployment status accordingly.
+	// In this case the status will contain the result from the first source
+	// which is still waiting to be unpacked.
+	for _, res := range unpackResult {
+		if res.State != source.StateUnpacked {
+			return &res, nil
+		}
+	}
+	// TODO: capture the list of resolved sources for all the successful entry points.
+	return &source.Result{State: source.StateUnpacked, Message: "Successfully unpacked"}, nil
+}
+
+// validateContents validates if the unpacked bundle contents are of the right format.
+func (b *bundleDeploymentReconciler) validateContents(ctx context.Context, format v1alpha2.FormatType, store store.Store) error {
+	errs := make([]error, 0)
+	for _, validator := range b.validators {
+		if err := validator.Validate(ctx, format, store); err != nil {
+			errs = append(errs, err)
+		}
+	}
+	return apimacherrors.NewAggregate(errs)
+}
+
+// deployContents calls the registered deployer to apply the bundle contents onto the cluster.
+func (b *bundleDeploymentReconciler) deployContents(ctx context.Context, store store.Store, bd *v1alpha2.BundleDeployment) (*deployer.Result, error) {
+	return b.deployer.Deploy(ctx, store, bd)
+}
```

Comment:
This function seems unneccessary. We could just use the `b.deployer.Deploy()` function directly